### PR TITLE
Operations handler - add status validation

### PR DESCRIPTION
**Pull request by mflilian**
### What :question: 
It was implemented on on ListOperations method, a request validation for the status parameter passed on the query. 

### Why 🤔 
Given the need to filter by operation's status, we have to make additional updates on Operations handler.

### CODE REVIEW

File: internal/api/handlers/operations_handler.go [hunk 1]
```diff
@@ -64,6 +64,12 @@ func (h *OperationsHandler) ListOperations(ctx context.Context, request *api.Lis
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	operationStatus := request.Status
+	if err != nil {
+		handlerLogger.Error(fmt.Sprintf("error parsing sorting parameters, orderBy: %+v", request.Status), zap.Error(err))
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
```
Comment (arthur29) [inline internal/api/handlers/operations_handler.go:71]: Why did this error validation is being done after an assignment to `operation.Status`? What do you think to move it close to which it was assigned?

File: internal/api/handlers/operations_handler.go [hunk 2]
```diff
@@ -103,11 +109,21 @@ func (h *OperationsHandler) ListOperations(ctx context.Context, request *api.Lis
...
+	case "finished":
+		return &api.ListOperationsResponse{FinishedOperations: finishedOperationResponse}, nil
+
+	default:
+		return &api.ListOperationsResponse{
+			PendingOperations:  pendingOperationResponse,
+			ActiveOperations:   activeOperationResponses,
+			FinishedOperations: finishedOperationResponse,
+		}, nil
+	}
```
Comment (arthur29) [inline internal/api/handlers/operations_handler.go:126]: How are you planning to handle the operations on error status?

File: proto/api/v1/operations.proto [hunk 3]
```diff
@@ -41,6 +41,8 @@ message ListOperationsRequest {
   // Default value is `created_at desc`
   // NOTE: On http protocol, this operates as a query parameter.
   string order_by = 2;
+  // Optional parameter for enabling filter by operation status
+  string status = 3;
```
Comment (arthur29) [inline proto/api/v1/operations.proto:45]: AFAIK the status is not a required parameter as `page` and `per_page`, what do you think to add the `optional` parameter? This link could help you https://developers.google.com/protocol-buffers/docs/proto#optional

**Review by arthur29 — CHANGES_REQUESTED** [review]

File: proto/api/v1/operations.proto [hunk 4]
```diff
@@ -41,6 +41,8 @@ message ListOperationsRequest {
   // Default value is `created_at desc`
   // NOTE: On http protocol, this operates as a query parameter.
   string order_by = 2;
+  // Optional parameter for enabling filter by operation status
```
Comment (guilhermocc) [inline proto/api/v1/operations.proto:44]:
```suggestion
  // Optional parameter for enabling filter by operation execution stage, can be one of [pending, active, history]
```

**Pull request closed without merging**